### PR TITLE
fix: optimize on-chain USD price fetching

### LINKED ISSUE #1066: Optimize USD price fetching
#### Description

Optimize on-chain USD price fetching to prevent laggy UI

#### Definition of done
- [ ] Description is created (Task creator)
- [ ] Code is complete (Developer)
- [ ] Scope of test and what was done section is defined in task (Developer)
- [ ] Review passed (Code reviewer)
- [ ] Verified manually (QA)
- [ ] Test cases created/updated if necessary (QA)
- [ ] Change successfully merged
- [ ] Release notes added if needed

### PR DESCRIPTION
**Pull request by berteotti**
# Summary

Fixes #1066

Optimize on-chain USD price fetching and prevent laggy UI

  # To Test

1. Open the page `swap`
2. Simulate a trade
3. Open settings modal
- [ ] Verify if changing settings isn't laggy
4. Use the app while simulating the trade
- [ ] Verify if using the app in general isn't laggy

### CODE REVIEW

**Comment by MilanVojnovic95** [comment]
Tested and looks ok to me 👍

**Review by Diogomartf — APPROVED** [review]

File: src/hooks/useUSDValue.ts [hunk 1]
```diff
@@ -7,23 +19,26 @@ import { tryParseAmount } from '../state/swap/hooks'
...
+  [ChainId.XDAI]: { token: USDC[ChainId.XDAI], platform: UniswapV2RoutablePlatform.SUSHISWAP },
 }
 
 const FETCH_PRICE_INTERVAL = 15000
 
-export function useUSDPrice(currencyAmount?: CurrencyAmount, selectedTrade?: Trade) {
+export function useUSDPrice(currencyAmount?: CurrencyAmount) {
   const { chainId } = useActiveWeb3React()
-  const stablecoin = chainId ? STABLECOIN_OUT[chainId] : undefined
+  const { stablecoin, platform }: { stablecoin?: Token; platform?: UniswapV2RoutablePlatform } = chainId
```
[the remark this answers is not in the document]
Comment (berteotti) [reply src/hooks/useUSDValue.ts]: As we have an empty object if there is no `chainId`, linter was throwing any type error.  Fixed in [0357269](https://github.com/levelkdev/swapr-dapp/pull/1067/commits/0357269286710fa684ecb0550ea6e799bc9e03f1)

File: src/hooks/useUSDValue.ts [hunk 2]
```diff
@@ -7,23 +19,26 @@ import { tryParseAmount } from '../state/swap/hooks'
...
+    ? STABLECOIN_AND_PLATFOM_BY_CHAIN[chainId]
+    : {}
 
-  const tradeExactOutAllPlatforms = useTradeExactInAllPlatforms(currencyAmount, stablecoin)
+  const tradeExactOutAllPlatforms = useTradeExactInUniswapV2(currencyAmount, stablecoin, platform)
```
[the remark this answers is not in the document]
Comment (berteotti) [reply src/hooks/useUSDValue.ts]: Fixed in [0357269](https://github.com/levelkdev/swapr-dapp/pull/1067/commits/0357269286710fa684ecb0550ea6e799bc9e03f1)

File: src/hooks/useUSDValue.ts [hunk 3]
```diff
@@ -7,23 +19,26 @@ import { tryParseAmount } from '../state/swap/hooks'
...
-import { useTradeExactInAllPlatforms } from './Trades'
+import { useTradeExactInUniswapV2 } from './Trades'
 
 import { useActiveWeb3React } from './index'
 
-const STABLECOIN_OUT: { [chainId: number]: Token } = {
-  [ChainId.MAINNET]: DAI,
-  [ChainId.ARBITRUM_ONE]: USDC[ChainId.ARBITRUM_ONE],
-  [ChainId.XDAI]: USDC[ChainId.XDAI],
+const STABLECOIN_AND_PLATFOM_BY_CHAIN: { [chainId: number]: { token: Token; platform: UniswapV2RoutablePlatform } } = {
```
[the remark this answers is not in the document]
Comment (berteotti) [reply src/hooks/useUSDValue.ts]: Fixed in [0357269](https://github.com/levelkdev/swapr-dapp/pull/1067/commits/0357269286710fa684ecb0550ea6e799bc9e03f1). Thanks for the tip, didn't know about this, much better!

**Pull request merged**